tag.go

```diff
@@ -69,6 +69,11 @@ func getLatestAcceptableTag(tagConstraint string, tags []string) (string, *Fetch
 		}
 	}
 
+    // check contstraint against only the latest acceptable version
+    if !constraints.Check(latestAcceptableVersion) {
+        return latestTag, wrapError(errors.New("Tag does not exist"))
+    }
+
```

Comment:
Ah, crap, right, `fetch-test-private` is a repo we intentionally made private to test that `fetch` works with private repos, so the test is failing because you don't have access. Just added you.